### PR TITLE
Upgrade to stackgres 1.8 and zfs  to 2.4.1

### LINKED ISSUE #7619: Upgrade to ZFS Local PV to 2.4.1
### Problem

The upstream [release](https://github.com/openebs/zfs-localpv/commits/zfs-localpv-2.4.1/) contains fixes that allow us to not use a custom image for our helm chart. We should upgrade to this version

### Solution

- [ ] update helm chart to latest version
- [ ] Deploy the common chart + V2 enabled 
- [ ] Verify nodes are labeled correctly
- [ ] scale nodepools to 0 and back up again. Verify that the citus shards are able to schedule and find the appropriate volumes

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by jnels124**
**Description**:
* upgrade stackgres to 1.8
* upgrade custom zfs image to latest

**Related issue(s)**:

Fixes #7619 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: charts/hedera-mirror-common/Chart.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ dependencies:
     condition: stackgres.enabled
     name: stackgres-operator
     repository: https://stackgres.io/downloads/stackgres-k8s/stackgres/helm/
-    version: 1.7.0
+    version: 1.8.0
```
Comment (steven-sheehy) [inline charts/hedera-mirror-common/Chart.yaml:25]: Need to also update workflow

File: charts/hedera-mirror-common/values.yaml [hunk 2]
```diff
@@ -481,4 +483,4 @@ zfs:
       registry: gcr.io/
       repository: mirrornode/zfs-driver
       pullPolicy: IfNotPresent
-      tag: 2.3.1-HEDERA
+      tag: 2.4.1-HEDERA
```
Comment (steven-sheehy) [inline charts/hedera-mirror-common/values.yaml:486]: Still can't use the official image it looks like due to a follow up fix?

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: charts/hedera-mirror-common/values.yaml (hunk 2 above)
Comment (jnels124) [reply charts/hedera-mirror-common/values.yaml:486]: Correct. There was an additional commit that is missing from the upstream PR that was merged. Originally I thought maybe it was lost when I rebased as they requested (still a possibility) but looking at the way I created the "release" branch I think it was the rebase done after mine that may have removed it. Anyways will get the fix in upstream.

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]

**Pull request merged**